npm/src/auth/index.ts

```diff
@@ -33,6 +33,26 @@ const isRefreshTokenExpired: (
   return token != undefined && token.exp != undefined && token.exp < now
 }
 
+export const refreshIdleSessionTimeout: (
+  keycloak: KeycloakInstance,
+  idleSessionMinValiditySecs?: number
+) => void = (keycloak, idleSessionMinValiditySecs = 60) => {
+  const refreshToken = keycloak.refreshTokenParsed
+  if (refreshToken != undefined && refreshToken.exp != undefined) {
+    const nowInSecs = Math.round(new Date().getTime() / 1000)
+    const expInSecs = refreshToken.exp
+    //Expiry is a future time, add min validity to the 'now' to check if expiry is about to expire
+    const timeoutInMs =
+      (expInSecs - (nowInSecs + idleSessionMinValiditySecs)) * 1000
+
+    setTimeout(() => {
+      refreshOrReturnToken(keycloak).then(() =>
+        refreshIdleSessionTimeout(keycloak)
```

Comment:
When we come to implement the session refresh modal, we'll need to have a discussion about whether this should run indefinitely, or if there should be some point at which even this process gets stopped.